### PR TITLE
Include MaxIdle field on client EntryView responses

### PR DESCRIPTION
**Pull request by tkountis**
Fixes https://github.com/hazelcast/hazelcast-client-protocol/issues/150

### CODE REVIEW

File: hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapBasicTest.java [hunk 1]
```diff
@@ -1076,8 +1077,29 @@ public void testGetEntryView() {
 
         assertEquals(key, view.getKey());
         assertEquals(value, view.getValue());
+        assertEquals(Long.MAX_VALUE, view.getMaxIdle());
```
Comment (blazember) [inline hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapBasicTest.java]: Looks like these two are the only two methods test the methods with `maxIdle` support. Shouldn't be more here?

Also, these are unrelated to this PR though:
- `testGetAll_whenMapEmpty` is missing `@Test` annotation
- many tests declare thrown exceptions while no invoked method is throwing checked exceptions

Comment (tkountis) [reply hazelcast-client/src/test/java/com/hazelcast/client/map/ClientMapBasicTest.java]: Thanks @blazember. 
I fixed the missing annotation & the exceptions.
Regarding the tests, I added 1 more to handle the map config value (when user gives negative input), but these are the tests handling `maxIdle` from an entry-view perspective. We test entries with maxIdle in the `EvictionTest` for both OSS/EE & client.

**Comment by tkountis** [comment]
run-lab-run
